biz.aQute.resolve/src/biz/aQute/resolve/ResolveProcess.java

```diff
@@ -209,6 +210,27 @@ public static ResolutionException augment(ResolveContext context, ResolutionExce
 		return augment(re.getUnresolvedRequirements(), context, re);
 	}
 
+	public static String format(Collection<Requirement> requirements) {
+		try (Formatter f = new Formatter()) {
+			for (Requirement req : requirements) {
+				boolean optional = isOptional(req);
+				f.format("%n %s[%-19s] %s%s", isOptional(req) ? " " : "*", req.getNamespace(), req,
```

Comment:
will try grouping but there was no "plan" because nobody said they wanted grouping!

As for `*`, as a heavy user of resolving, this tiny hint is a huge usability improvement once you realize what it means, which takes a slit second to recognize, but grouping should also not require it.